nix/haskell.nix

```diff
@@ -1,12 +1,25 @@
 self: super:
 
-{ haskellPackages = super.haskellPackages.override 
-  (old:
-    { overrides = super.lib.composeExtensions (old.overrides or (_: _: {}))
-      ( hself: hsuper:
-        { optparse-repline = hself.callCabal2nix "optparse-repline" ../. {};
-        }
-      );
-    }
-  );
+{
+  haskellPackages = super.haskellPackages.override (old: {
+    overrides = super.lib.composeExtensions (old.overrides or (_: _: { }))
+      (hself: hsuper: {
+        haskeline = super.haskell.lib.dontCheck (hself.callCabal2nix "haskeline"
+          (builtins.fetchTarball {
+            url =
+              "https://hackage.haskell.org/package/haskeline-0.8.1.0/haskeline-0.8.1.0.tar.gz";
+            sha256 =
+              "sha256:1r33spdza6mj9qfrsv08sysnkvdz7l8qdhwa5yhq2f9my1ikg85x";
```

Comment:
Is manually setting the repo to grab the library from going to break any upstream logic for compatible dependencies (immediate and transient)?

Reply:
No, this should be fine: the only reason to do it this way is that the latest `haskeline`/`repline` packages weren't in the all-hashes set for the pinned nixpkgs, so this just pulls the source directly from hackage. 

If/when the nixpkgs pin gets updated, this can be removed/tweaked as appropriate.

Reply:
Makes sense!